queries.sql

```diff
@@ -81,3 +81,71 @@ WHERE  date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
 GROUP BY species;
 
 
+
+
+SELECT  full_name, name
+   FROM animals
+   INNER JOIN owners
+   ON animals.owners_id = owners.id
+WHERE owners_id = 4;
+
+
+SELECT  animals.name, species.name
+   FROM animals
+   INNER JOIN species
+   ON animals.species_id = species.id
+WHERE species.name = 'Pokemon';
+  
+
+SELECT  full_name, name
+   FROM animals
+   RIGHT JOIN owners
+   ON animals.owners_id = owners.id;
+
+
+SELECT  species.name, count(animals.name)
+   FROM animals
+   INNER JOIN species
+   ON animals.species_id = species.id
+GROUP BY species.name;
+
+
+SELECT  species.name, count(animals.name)
+   FROM animals
+   INNER JOIN species
+   ON animals.species_id = species.id
+GROUP BY species.name;
+
+
+SELECT  animals.name, full_name, species.name as species_name
+   FROM animals
+   INNER JOIN owners 
+   ON animals.owners_id = owners.id
+   INNER JOIN species 
+   ON animals.species_id = species.id
+where full_name = 'Jennifer Orwell' and species.name = 'Digimon';
+
+
+SELECT  animals.name, full_name, escape_attempts
+   FROM animals
+   INNER JOIN owners 
+   ON animals.owners_id = owners.id
+   INNER JOIN species 
+   ON animals.species_id = species.id
+where full_name = 'Dean Winchester' and escape_attempts = 0;
+
+
+
+INSERT INTO result (full_name, num ) SELECT full_name, COUNT(animals.name)
+     FROM animals
+     INNER JOIN owners 
+     ON animals.owners_id = owners.id
+  GROUP BY full_name;
+
+
+SELECT  full_name, num
+    from result
+    where num = (SELECT MAX(num) FROM result );
+
```

Comment:
- [OPTIONAL] In further reference to my comment ..., there is no need for the result table. If you notice, you are making performing several actions in the database and that is less efficient and puts more load on the database.  All you need to do is select the owners' names, and the count of animals that they have, order the count in descending order and have the result limited to one. 😃 😎 🌵 

#### Hint: 

```SQL
SELECT full_name, COUNT(animals.name)
   FROM animals
   INNER JOIN owners 
   ON animals.owners_id = owners.id
   GROUP BY full_name order by COUNT(animals.name) DESC LIMIT 1;
```